components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2892,9 +2893,9 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
 
     public async getGitpodTokens(ctx: TraceContext): Promise<GitpodToken[]> {
         const user = await this.checkAndBlockUser("getGitpodTokens");
-        const res = (await this.userDB.findAllGitpodTokensOfUser(user.id)).filter((v) => !v.deleted);
-        await Promise.all(res.map((tkn) => this.guardAccess({ kind: "gitpodToken", subject: tkn }, "get")));
-        return res;
+        return this.gitpodTokenService.getGitpodTokens(user.id, user.id, (token: GitpodToken) => {
+            return this.guardAccess({ kind: "gitpodToken", subject: token }, "get");
```

Comment:
This one could be checked entirely here after the tokens got returned. No need to pass in the oldPErmissionCheck lambda.
I,.e. just keep the old code as is:
```
await Promise.all(res.map((tkn) => this.guardAccess({ kind: "gitpodToken", subject: tkn }, "get")));
```